hedera-node/hedera-app/src/main/java/com/hedera/node/app/roster/schemas/V057RosterSchema.java

```diff
@@ -74,10 +83,17 @@ public void restart(@NonNull final MigrationContext ctx) {
         if (!ctx.configuration().getConfigData(AddressBookConfig.class).useRosterLifecycle()) {
             return;
         }
-        final var rosterStore = rosterStoreFactory.apply(ctx.newStates());
+        final var states = ctx.newStates();
+        final var tssStore = tssStoreFactory.apply(states);
```

Comment:
I think here we might have the same problem (mentioned [here](https://github.com/hashgraph/hedera-services/pull/16638#discussion_r1852220819)) that this schema can't "see" states owned by `TssBaseService`. 🤔  
@tinker-michaelj please correct me if the case here is somewhat different.

Reply:
Updated with a supplier for the tss store when creating RosterService

Reply:
Agreed, we generally try to avoid cross-service access. Why do we want to remove the TSS encryption key if a node ID isn't in both the active and candidate rosters?

Reply:
@mhess-swl 
I assume to not overload the state with unnecessary data in regards to the current active and candidate rosters. In the design document it is just mentioned about the cleanup but not the reasoning behind..

Reply:
I agree, the approach is changed and now TSS schema reads from roster store and update it's state, if necessary